src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -545,16 +544,11 @@ private EvaluationSubmitter getEvaluationSubmitter() {
     return submitter;
   }
 
-  private UploadDialogWidget getNewUploadDialogWidget() {
-    UploadDialogWidget uploadDialogWidget = ginInjector.getUploadDialogWidget();
-    view.setUploadDialogWidget(uploadDialogWidget.asWidget());
-    return uploadDialogWidget;
-  }
-
-  private UploadDialogWidgetV2 getNewUploadDialogWidgetV2() {
-    UploadDialogWidgetV2 uploadDialogWidgetV2 =
-      ginInjector.getUploadDialogWidgetV2();
-    view.setUploadDialogWidget(uploadDialogWidgetV2.asWidget());
+  private UploadDialogWidgetV2 getUploadDialogWidget() {
+    if (uploadDialogWidgetV2 == null) {
+      uploadDialogWidgetV2 = ginInjector.getUploadDialogWidget();
+      view.setUploadDialogWidget(uploadDialogWidgetV2.asWidget());
+    }
```

Comment:
Use a single instance of the widget to prevent bugs where the widget instance that was configured for drag-and-drop was not the same instance of the widget that is shown to the user. I don't think the present code would have this bug, but I introduced it in an earlier iteration and it would otherwise be very easy to reintroduce in a later refactor

Reply:
We should do some extra testing with component initialization with this change.  After uploading a file (or set of files), if you open the uploader to upload another set of files is the component state properly re-initialized?